### PR TITLE
Relax test assumptions

### LINKED ISSUE #18: Strong assumptions about form of distributions in TestUtils
The `Models.TestUtils` tests currently assume that single-output regressors return vectors of `Normal` distributions, and that multi-output regressors return `Vector`s of `MvNormal`s. See [here](https://github.com/invenia/Models.jl/blob/9fa0caa695f621baeb6953fd354413b54b85339a/src/test_utils.jl#L161) and [here](https://github.com/invenia/Models.jl/blob/9fa0caa695f621baeb6953fd354413b54b85339a/src/test_utils.jl#L171) resp.

@glennmoy @nickrobinson251 why are we doing this? It seems overly-restrictive.

### PR DESCRIPTION
**Pull request by willtebbutt**
Resolves #18

### CODE REVIEW

File: src/test_utils.jl [hunk 1]
```diff
@@ -88,7 +88,7 @@ distribution (with zero-vector mean and identity covariance matrix) for each obs
 function FakeTemplate{DistributionEstimate, MultiOutput}()
     FakeTemplate{DistributionEstimate, MultiOutput}() do num_variates, inputs
         std_dev = ones(num_variates)
-        return [MultivariateNormal(std_dev) for _ in 1:size(inputs, 2)]
+        return [Product(Normal.(0, std_dev)) for _ in 1:size(inputs, 2)]
```
Comment (oxinabox) [inline src/test_utils.jl:91]: do we need to change the fakes?
I worry that some compositional models do require that the component is a gaussian.
And that then we won't be able to use the fakes to test them anymore.

File: src/test_utils.jl [hunk 2]
```diff
@@ -168,7 +168,7 @@ function test_interface(
     inputs=rand(5, 5), outputs=rand(3, 5)
 )
     predictions = test_common(template, inputs, outputs)
-    @test predictions isa Vector{<:MultivariateNormal{<:Real}}
+    @test predictions isa Vector{<:ContinuousMultivariateDistribution}
```
Comment (oxinabox) [inline src/test_utils.jl]: maybe while we are here we can relax these to `AbstractVector`s ?

File: src/test_utils.jl (hunk 1 above)
Comment (willtebbutt) [reply src/test_utils.jl:91]: Which compositional models?

Comment (willtebbutt) [reply src/test_utils.jl:91]: > do we need to change the fakes?

Our Fakes would have allowed for this bug fix to regress. Making the fakes some random distribution is a good idea in that regard.

Comment (oxinabox) [reply src/test_utils.jl:91]: Actually non will do that becuase a distribution is never an input, i think it is fine

**Review by oxinabox — APPROVED** [review]

File: src/test_utils.jl [hunk 3]
```diff
@@ -168,7 +168,7 @@ function test_interface(
     inputs=rand(5, 5), outputs=rand(3, 5)
 )
     predictions = test_common(template, inputs, outputs)
-    @test predictions isa Vector{<:MultivariateNormal{<:Real}}
```
Comment (nickrobinson251) [inline src/test_utils.jl:171]: Why do these have to be `Continuous`?

Comment (oxinabox) [reply src/test_utils.jl:171]: Because multivariate classification isn't a thing we care about supporting?
We barely support classification.

**Pull request merged**